### PR TITLE
installation: document installing shell auto-completion

### PR DESCRIPTION
**Pull request by oliver-sanders**
Document installing the Bash auto-completion script.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

File: src/installation.rst [hunk 1]
```diff
@@ -281,3 +285,30 @@ configuration in login scripts.
...
+^^^^^^^^^^^^^^^^^^^^^
+
+Cylc provides auto-completion for the Bash shell which can save you typing:
+
+* Cylc commands
+* Workflow IDs
+* Cycle points
+* Task names
+
+To extract the auto-completion file run the folling command:
```
Comment (datamel) [inline src/installation.rst:301]:
```suggestion
To extract the auto-completion file run the following command:
```

**Review by datamel — COMMENTED** [review]
Small typo. Would it be worth updating the `cylc get-resources --help` to include an example of the bash completion? Or is the intention that on the whole sites will be doing this for users?

**Comment by oliver-sanders** [comment]
There's an entry in the `--list`:

```console
$ cylc get-resources --list
Resources:
  syntax/cylc-mode.el     # Emacs syntax highlighting.
  syntax/cylc.lang        # Gedit (gtksourceview) syntax highlighting.
  syntax/cylc.vim         # Vim syntax highlighting.
  syntax/cylc.xml         # Kate syntax highlighting.
  cylc-completion.bash    # Bash auto-completion for Cylc commands.
  cylc                    # Cylc wrapper script.
```

**Review by wxtim — APPROVED** [review]

**Comment by wxtim** [comment]
Legit & Small & you've responded to @datamel 's comment.

**Pull request merged**